dex-it/src/test/scala/com/wavesplatform/it/sync/CancelOrderTestSuite.scala

```diff
@@ -20,14 +27,90 @@ class CancelOrderTestSuite extends MatcherSuiteBase {
     xs.foreach(tx => node.waitForTransaction(tx.id))
   }
 
+  def createAccountWithBalance(balances: (Long, Option[String])*): KeyPair = {
+    val account = KeyPair(ByteStr(s"account-test-${System.currentTimeMillis}".getBytes(StandardCharsets.UTF_8)))
+
+    balances.foreach {
+      case (balance, asset) => {
+        if (asset != None)
+          assert(
+            node.assetBalance(alice.toAddress.toString, asset.get.toString).balance >= balance,
+            s"Alice doesn't have enough balance in ${asset.get.toString} to make a transfer"
+          )
+        node.waitForTransaction(node.broadcastTransfer(alice, account.toAddress.toString, balance, 300000L, asset, None).id)
+      }
+    }
+    account
+  }
+
   "Order can be canceled" - {
 
+    "After cancelAllOrders all of them should be cancelled (ASYNC)" in {
+      val account1 = createAccountWithBalance(100000000000L -> None)
+      val account2 = createAccountWithBalance(100000000000L -> None)
+      val account3 = createAccountWithBalance(100000000000L -> None)
+      val account4 = createAccountWithBalance(100000000000L -> None)
+      val account5 = createAccountWithBalance(100000000000L -> None)
+      val account6 = createAccountWithBalance(100000000000L -> None)
+
+      SyncMatcherHttpApi.sync(
+        {
+          import com.wavesplatform.it.api.AsyncMatcherHttpApi.{MatcherAsyncHttpApi => async}
+
+          val asyncNode = async(node)
+
+          def place(account: KeyPair, startPrice: Long): IndexedSeq[Future[MatcherResponse]] = {
+            val time = System.currentTimeMillis
+
+            val futures = for {
+              c <- 1 until 199
+            } yield {
+              Thread.sleep(50)
+              asyncNode.placeOrder(account, wavesUsdPair, SELL, 100000000L, startPrice + c, 300000L, 2.toByte, timestamp = time + c)
+            }
+
+            futures
+          }
+
+          def cancelAll(account: KeyPair): Unit = {
+            asyncNode.cancelAllOrders(account, System.currentTimeMillis)
+          }
+
+          val p1 = place(account1, 1000)
+          val p2 = place(account2, 2000)
+          val p3 = place(account3, 3000)
+          cancelAll(account1)
+          cancelAll(account2)
+          val p4 = place(account4, 4000)
+          val p5 = place(account5, 5000)
+          cancelAll(account3)
+          val p6 = place(account6, 6000)
+          cancelAll(account4)
+          cancelAll(account5)
+          cancelAll(account6)
+
+          Future.sequence(p1)
+          Future.sequence(p2)
+          Future.sequence(p3)
+          Future.sequence(p4)
+          Future.sequence(p5)
+          Future.sequence(p6)
+        },
```

Comment:
Only a Future in the last expression will be waited for 3 minutes. Future.sequence doesn't block thread. The right version could be:
```scala
Future
.sequence(List(
  place(account1, 1000),
  place(account2, 2000),
  place(account3, 3000),
  cancelAll(account1),
  cancelAll(account2),
  place(account4, 4000),
  place(account5, 5000),
  cancelAll(account3),
  place(account6, 6000),
  cancelAll(account4),
  cancelAll(account5),
  cancelAll(account6)
))
.flatMap(_ => Future.sequence(List(account1, account2, account3, account4, account5, account6).map(cancelAll)))
```

I'll fix later.